src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -1039,6 +1041,212 @@ class npc_magister_aledis : public CreatureScript
     }
 };
 
+enum WatchCommanderLeonus
+{
+    SAY_COVER                   = 0,
+    EVENT_LEONUS_TALK           = 1,
+    EVENT_INFERNAL_RAIN_ATTACK  = 2,
+    EVENT_FEAR_CONTROLLER_CAST  = 3,
+    EVENT_ACTIVE_FALSE          = 4,
+    NPC_INFERNAL_RAIN           = 18729,
+    SPELL_INFERNAL_RAIN         = 33814,
+    NPC_FEAR_CONTROLLER         = 19393,
+    DATA_ACTIVE                 = 1,
+};
+
+struct npc_watch_commander_leonus : public ScriptedAI
+{
+    npc_watch_commander_leonus(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _events.ScheduleEvent(EVENT_LEONUS_TALK, 2min, 10min);
+        _events.ScheduleEvent(EVENT_INFERNAL_RAIN_ATTACK, 2min, 10min);
+        _events.ScheduleEvent(EVENT_FEAR_CONTROLLER_CAST, 2min, 10min);
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        switch (data)
+        {
+            case 1:
+                _events.ScheduleEvent(EVENT_ACTIVE_FALSE, 1s);
+                break;
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_LEONUS_TALK:
+                    Talk(SAY_COVER);
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_SHOUT);
+                    break;
+                case EVENT_INFERNAL_RAIN_ATTACK:
+                {
+                    std::list<Creature*> infernalrainList;
+                    Trinity::AllCreaturesOfEntryInRange checkerInfernalrain(me, NPC_INFERNAL_RAIN, 200.0f);
+                    Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcherInfernal(me, infernalrainList, checkerInfernalrain);
+                    Cell::VisitAllObjects(me, searcherInfernal, 200.0f);
+
+                    if (infernalrainList.empty())
```

Comment:
not needed
if empty will skip the loop (in line 1102)

Reply:
@Kittnz what about these empty checks ?